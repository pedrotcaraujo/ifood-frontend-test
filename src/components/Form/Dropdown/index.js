import React, { Component } from 'react';

class Dropdown extends Component {
    render() {
        return (
            <div>
                {this.props.label && <label>{this.props.label}</label>}
                <select {...this.props.dropdownProps}>
                    {this.props.options.map(option => (
                       <option key={option.value} value={option.value}>{option.name}</option>
                    ))}
                </select>
            </div>
        )
    }
}

export default Dropdown;