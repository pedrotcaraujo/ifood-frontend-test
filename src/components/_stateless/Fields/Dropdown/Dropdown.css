.Dropdown {
    margin: 5px 10px;
    width: 100%;
}

.Dropdown-field {
    border: none;
    border-radius: 2px;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.16), 0 0 0 1px rgba(0,0,0,0.08);
    height: 44px;
    outline: none;
    background: #fff;
    box-sizing: border-box;
    margin: 0;
    padding: 0 8px;
    width: 100%;
    font-size: 1em;
    font-weight: bold;
}