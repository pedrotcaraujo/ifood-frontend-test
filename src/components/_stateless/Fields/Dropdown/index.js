import React, { Component } from 'react';
import './Dropdown.css';
import randomString from '../../../../utils/randomString'

class Dropdown extends Component {
    render() {
        return (
            <div className="Dropdown">
                {this.props.label && <label className="Dropdown-label">{this.props.label}</label>}
                <select className="Dropdown-field" {...this.props.dropdownProps}>
                    {this.props.options.map(option => (
                       <option key={randomString(16)} value={option.value}>{option.name}</option>
                    ))}
                </select>
            </div>
        )
    }
}

export default Dropdown;