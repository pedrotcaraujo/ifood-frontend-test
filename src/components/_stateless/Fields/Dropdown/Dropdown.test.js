import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Dropdown from './index';

const optionsMock = [
    {
        name: 'teste_name_1',
        value: 'teste_value_1'
    },
    {
        name: 'teste_name_2',
        value: 'teste_value_2'
    }
];


it('renders correctly', () => {
  const tree = renderer
    .create(<Dropdown options={optionsMock} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Dropdown options={optionsMock} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
