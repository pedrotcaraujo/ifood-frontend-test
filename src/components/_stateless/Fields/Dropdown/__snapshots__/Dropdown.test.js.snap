// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Dropdown"
>
  <select
    className="Dropdown-field"
  >
    <option
      value="teste_value_1"
    >
      teste_name_1
    </option>
    <option
      value="teste_value_2"
    >
      teste_name_2
    </option>
  </select>
</div>
`;
